addons/mrp/models/stock_orderpoint.py

```diff
@@ -88,3 +88,13 @@ def _prepare_procurement_values(self, date=False, group=False):
         values = super()._prepare_procurement_values(date=date, group=group)
         values['bom_id'] = self.bom_id
         return values
+
+    def _post_process_scheduler(self):
+        """ Confirm the productions only after all the orderpoints have run their
+        procurement to avoid the new procurement created from the production conflict
+        with them. """
+        self.env['mrp.production'].search([
```

Comment:
sudo() ?

Reply:
Also I am afraid about `with_user(SUPERUSER_ID).sudo().with_company(company_id)` missing in this confirm

Reply:
You're right. The company is already managed correctly but not the access rights